controllers/validator.go

```diff
@@ -23,6 +23,9 @@ func (r *DPAReconciler) ValidateDataProtectionCR(log logr.Logger) (bool, error)
 		return false, errors.New("DPA CR Velero configuration cannot be nil")
 	}
 
+	if dpa.Spec.Configuration.Restic.Enable != nil && *dpa.Spec.Configuration.Restic.Enable {
+		dpa.Spec.Configuration.Velero.Args.RestoreResourcePriorities = "ServiceAccount,SecurityContextConstraints"
+	}
```

Comment:
Thank you @sseago. I will add the condition to change only if RestoreResourcePriorities field is currently empty.

Reply:
Where does this complete list of priorities come from?

Reply:
@shawn-hurley https://github.com/vmware-tanzu/velero/blob/main/pkg/cmd/server/server.go#L543

Reply:
@shawn-hurley if user supplies a priority list, it replaces the default, which means if we want to add SCC to the default, we need to supply the full list with SCC added in the right place.

Reply:
I actually think it's probably better to swap the order of securitycontextconstraints and customresourcedefinitions in the list. In principle, CRDs should be restored before any CRs, even though in practice it shouldn't matter for an OpenShift cluster, since the SCC CRD will already be defined.

Reply:
Is it possible to format the string one entry per line so the priorities are easier to read? 
e.g.
```
 543 var defaultRestorePriorities = restore.Priorities{
 544         HighPriorities: []string{
 545                 "customresourcedefinitions",
 546                 "namespaces",
 547                 "storageclasses",
 548                 "volumesnapshotclass.snapshot.storage.k8s.io",
 549                 "volumesnapshotcontents.snapshot.storage.k8s.io",
 550                 "volumesnapshots.snapshot.storage.k8s.io",
 551                 "datauploads.velero.io",
 552                 "persistentvolumes",
 553                 "persistentvolumeclaims",
 554                 "serviceaccounts",
 555                 "secrets",
 556                 "configmaps",
 557                 "limitranges",
 558                 "pods",
 559                 // we fully qualify replicasets.apps because prior to Kubernetes 1.16, replicasets also
 560                 // existed in the extensions API group, but we back up replicasets from "apps" so we want
 561                 // to ensure that we prioritize restoring from "apps" too, since this is how they're stored
 562                 // in the backup.
 563                 "replicasets.apps",
 564                 "clusterclasses.cluster.x-k8s.io",
 565                 "endpoints",
 566                 "services",
 567         },
 568         LowPriorities: []string{
 569                 "clusterbootstraps.run.tanzu.vmware.com",
 570                 "clusters.cluster.x-k8s.io",
 571                 "clusterresourcesets.addons.cluster.x-k8s.io",
 572         },
```

Reply:
Is this doc'd somewhere wlese (as a follow on or something) such that if a user does want to change this, they at least know they must have certain things in certain order?

Otherwise this makes sense, I don't know why we have VMware resources TBH in oadp but 🤷🏻

Reply:
@weshayutin We could format it that way in a const and then parse or transform it into the format that velero needs, but what we eventually need to pass to velero is a comma-separated list string with a hyphen entry separating high and low priorities -- i.e. "high1,high2,-,low1,low2"

Reply:
@weshayutin It's a CLI param parsed by `velero server`. What you pasted above is the internal format that the CLI param gets parsed into.

Reply:
@shawn-hurley I don't believe it's documented -- that was one of the tasks we talked about in scrum -- we need to document for users that if they override this, they need to provide this list. That's actually a reason to keep it as-is in the code rather than something we transform into the cli string, so users have an example as to what their custom string should look like. Should the doc go in this PR (under docs/examples maybe?) or something added by the docs team in a docs PR later?

Reply:
@shawn-hurley " I don't know why we have VMware resources TBH in oadp but 🤷🏻" -- because the intent was to keep the same list that would be used by default with the one new entry added. I guess we could remove the vmware-specific ones if we're  certain that no OADP user will ever have these resources, but I think it's easier to maintain if our list is just "everything from default plus SCC".

Reply:
@sseago and @shawn-hurley  @hhpatel14 has started a doc pr here: https://github.com/openshift/oadp-operator/pull/1133   Listing the default in the FAQ would be start and something we could hand off to the docs team IMHO.

Reply:
@weshayutin That PR claims that it's only a temporary fix. What @shawn-hurley and I were talking about is a more permanent note -- any user who needs to set custom restore order will need to fit their own restore needs into the context of the default list. Should that PR be modified to handle the temporary "everyone needs this" wording and the more permanent "you still need this if you override stuff"?

Reply:
Also, the doc note that users overriding restore priorities need to replicate the full list isn't new to 4.14, it's always been the case. The difference now is that users must replicate the *OADP* default list instead of just the Velero one.

Reply:
Try to sum up the responses:

1. I agree that it is easier to make sure it is the same to keep vmware resource in the list. 
2. I agree with Scott, the doc of some type that says if you update the default list of resources, you must have SCC before XX or whatever the rules are.
3. Do we want to consider adding some validation to the field based on the above doc?

Reply:
@shawn-hurley  Validation might be helpful, although if we decide to go there, we may want to validate more than just SCC, but also the necessary priority for the resources that Velero relies on (i.e. pv before pvc before pod, etc.) -- problem is I don't think those relationships and needs are really documented anywhere other than their presence in the default list -- so maybe our validation would need to make sure that any DPA- supplied value here includes the full default velero list, in the proper order, with whatever custom changes added.

I guess we could take the string that we set as the OADP default and compare that to user-supplied value. User-supplied value must include everything from the default list, in the default order, with the only changes being that their own custom needs are added into this list somewhere.